torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1929,9 +2390,97 @@ def __init__(
         self.frozen = False
         self.dim_constraints: Optional[DimConstraints] = None
 
+        self.graph_added_nodes: Dict[Tuple[Callable, Tuple[Any, ...]], torch.fx.Node] = {}
+        self.source_to_symbol: Dict[str, sympy.Symbol] = {}
+        if _translation_validator_enabled():
+            self.validator = TranslationValidator()
+            self.graph = torch.fx.Graph()
+            # Create an output graph and start inserting before that.
+            # This is needed when 'deepcopy'-ing this object.
+            self.graph.inserting_before(self.graph.output(None))
+
     def freeze(self):
         self.frozen = True
 
+    def _create_symbol_for_source(self, source: Source, positive: Optional[bool]) -> Optional[sympy.Symbol]:
+        if _translation_validator_enabled():
+            srcname = source.name()
+            if source not in self.source_to_symbol:
+                self.source_to_symbol[srcname] = sympy.Symbol(srcname, positive=positive, integer=True)
+            return self.source_to_symbol[srcname]
+
+    def _add_z3var_for(self, symbol: sympy.Symbol, type: Type) -> None:
+        if _translation_validator_enabled():
+            if type == int:
+                self.validator.add_int(symbol)
+            elif type == float:
+                self.validator.add_real(symbol)
+            elif type == bool:
+                self.validator.add_bool(symbol)
+            else:
+                raise ValueError(f"can't create Z3 variable for type: {type}")
+
+    def _add_output_guard(self, expr) -> None:
+        if _translation_validator_enabled():
+            self.validator.add_output(expr)
+
+    def _add_assertion(self, expr) -> None:
+        if _translation_validator_enabled():
+            self.validator.add_assertion(expr)
+
+    def _check_translation_validate(self) -> None:
+        if _translation_validator_enabled():
+            result = self.validator.validate()
+            if not result.success:
+                assert result.model is not None and result.failed_inputs is not None and result.outputs is not None
+                oneline_model = {sym: result.model[sym] for sym in result.model}
+                failed_inputs = "\n".join(f"==>> {inp}" for inp in result.failed_inputs)
+                outputs = "\n".join(f"==>> {out}" for out in result.outputs)
+                raise RuntimeError(f"""translation validation failed with model: {oneline_model}
+Outputs:
+{outputs}
+
+Failed inputs:
+{failed_inputs}""")
+
+    def create_fx_call_function(
+            self,
+            op: Callable,
+            args: Tuple,
+    ) -> Optional[torch.fx.Node]:
+        # Cache this tuple in order to avoid dupplicated nodes.
+        node_key = (op, args)
+
+        if _translation_validator_enabled() and node_key not in self.graph_added_nodes:
+            # Presence of None in the arguments implies that we should ignore this operation.
```

Comment:
How come?

Reply:
There are some reasons for that:
- Inductor creates `SymInt` nodes using `ShapeEnv`
- As far as I understand, we don't want to keep track of those
- We are not dealing with `None`, so we can use that as a marker of "uninteresting nodes"

Reply:
Ah OK, I don't really understand the inductor fresh symints, but I agree this is a pain